src/main.h

```diff
@@ -1303,6 +1303,7 @@ class CBlockIndex
     // Gridcoin (17.6.2017 Brod) Add dpor related back-pointers
     CBlockIndex* ppCpid;  // previous block by same cpid
     CBlockIndex* ppSuper; // previous superblock
+    CSuperblock* pDesSuper; // deserialized superblock data

```

Comment:
Unfortunately, for one specific case of reward calculation the whole chain must be scanned on startup to calculate lifetime average magnitude and lifetime research payments of cpid. We can save this info however and use simple moving average to alter in case of reorg. Six months are needed for beacons and only two weeks for magnitude unit.